src/design/introspection.stanza

```diff
@@ -89,3 +90,89 @@ public defn get-element-ports (obj:JITXObject) -> [JITXObject, JITXObject] :
     (_:None):
       [obj.p[1], obj.p[2]]
 
+doc: \<DOC>
+Get the final name field for a ref
+
+This function will return the final field in a ref.
+
+@param r Ref object that we will interrogate
+@return The field field of the ref as either a VarRef or an IndexRef.
+
+@snippet Example #1
+
+```stanza
+val A = #R(some.mod.C.GND[0])
+val N = get-ref-name(A)
+println("N: %_" % [N])
+
+; Will Print:
+; N: GND[0]
+
+val B = #R(some.other.C.VDD)
+val N2 = get-ref-name(A)
+println("N2: %_" % [N2])
+
+; Will Print:
+; N2: VDD
+```
+<DOC>
+public defn get-ref-name (r:Ref) -> Ref:
+  ; IndexRef has a index value
+  ;  that we must compensate for.
+  ; Other wise, we just can take the last value.
+  val nameIndex = match(r):
+    (x:IndexRef): 2
+    (x): 1
+  tail(r, ref-length(r) - nameIndex)
+
+
+doc: \<DOC>
+Retrieve a mapping of Ports to Pads
+
+This function will retrieve all of the
+@param dev Instance of a `pcb-component`. This function odes not work for `pcb-module` instances.
+@return Tuple of KeyValue mappings between `Pin` and a Tuple of `Pad` objects.
+<DOC>
+public defn port-to-pads (dev:JITXObject) -> Tuple<KeyValue<JITXObject, Tuple<JITXObject>>> :
+  ;Get the landpattern definition of the instance's component.
+  val lp = landpattern(instance-definition(dev))
+
+  ;Build a table storing each landpattern pad by its ref.
+  val pad-table = to-hashtable<Ref, JITXObject>(seq({ref(_0) => _0}, pads(lp)))
+
+  ;Scroll through pin property table to map instance port to landpattern pads.
+  for pt in pins(dev) map :
+    val pads = for pad-ref in property(pt.pads) seq :
```

Comment:
To be safe, we should call `get-property?` here instead which returns a Maybe.
It'll be None for components that do not use a pin-properties table (and technically for modules too).